### PR TITLE
fix: sdk7 ui pointer blocking

### PR DESCRIPTION
**Pull request by pravusjif**
* Set non-pointer blocking by default
* Added forced exceptions on event-based elements (elements with pointer events, text input)
* Integrated new `PointerFilter` property of `UITransform` component
* Added tests

-------------------
Related PRs:
* https://github.com/decentraland/protocol/pull/180
* https://github.com/decentraland/js-sdk-toolchain/pull/792

----------------
### <samp>🤖 Generated by Copilot at dced3b8</samp>

Updated the `@dcl/protocol` dependency and added the `pointer_filter` field to the UI transform component, which allows specifying the pointer interaction mode for UI elements. Refactored and tested the logic for setting the `pickingMode` of the UI elements based on the `pointer_filter` value, to ensure consistent and expected behavior for UI pointer events.

---------------

## QA Test Instructions

1. Enter [this playground scene](https://playground.decentraland.org/?sdk-branch=feat%2Fui-pointer-blocker-property&explorer-branch=fix%2Fsdk7-ui-mouse-blocking&code=CgpSZWFjdEVjc1JlbmRlcmVyLnNldFVpUmVuZGVyZXIoKCkgPT4gKAogIDxVaUVudGl0eQogICAgdWlUcmFuc2Zvcm09e3sKICAgICAgd2lkdGg6IDQwMCwKICAgICAgaGVpZ2h0OiAyMzAsCiAgICAgIHBvc2l0aW9uVHlwZTogImFic29sdXRlIiwKICAgICAgYWxpZ25TZWxmOiAiY2VudGVyIiwKICAgICAgcG9zaXRpb246IHsKICAgICAgICBsZWZ0OiAiMzUlIgogICAgICB9CiAgICB9fQogICAgdWlCYWNrZ3JvdW5kPXt7CiAgICAgIGNvbG9yOiBDb2xvcjQuY3JlYXRlKDAuNSwgMC44LCAwLjEsIDAuNikKICAgIH19CiAgPgogICAgPFVpRW50aXR5CiAgICAgIHVpVHJhbnNmb3JtPXt7CiAgICAgICAgd2lkdGg6ICcxMDAlJywKICAgICAgICBoZWlnaHQ6ICcxMDAlJywKICAgICAgICBmbGV4RGlyZWN0aW9uOiAnY29sdW1uJywKICAgICAgICBhbGlnbkl0ZW1zOiAnY2VudGVyJywKICAgICAgICBqdXN0aWZ5Q29udGVudDogJ3NwYWNlLWJldHdlZW4nCiAgICAgIH19CiAgICAgIHVpQmFja2dyb3VuZD17eyBjb2xvcjogQ29sb3I0LmZyb21IZXhTdHJpbmcoIiM3MGFjNzZmZiIpIH19CiAgICA%2BCiAgICAgIDxVaUVudGl0eQogICAgICAgIHVpVHJhbnNmb3JtPXt7CiAgICAgICAgICB3aWR0aDogJzEwMCUnLAogICAgICAgICAgaGVpZ2h0OiA1MCwKICAgICAgICAgIG1hcmdpbjogJzhweCAwJywKICAgICAgICAgIHBvaW50ZXJGaWx0ZXI6IFBvaW50ZXJGaWx0ZXJNb2RlLlBGTV9CTE9DSwogICAgICAgIH19CiAgICAgICAgdWlCYWNrZ3JvdW5kPXt7CiAgICAgICAgICB0ZXh0dXJlTW9kZTogJ2NlbnRlcicsCiAgICAgICAgICBjb2xvcjogQ29sb3I0LlllbGxvdygpCiAgICAgICAgfX0KICAgICAgLz4KICAgICAgPExhYmVsCiAgICAgICAgdmFsdWU9eydCTE9DS1MgUE9JTlRFUid9CiAgICAgICAgZm9udFNpemU9ezE4fQogICAgICAgIHVpVHJhbnNmb3JtPXt7IAogICAgICAgICAgd2lkdGg6IDE1MCwgCiAgICAgICAgICBoZWlnaHQ6IDMwLCAKICAgICAgICAgIHBvaW50ZXJGaWx0ZXI6IFBvaW50ZXJGaWx0ZXJNb2RlLlBGTV9OT05FLAogICAgICAgICAgcG9zaXRpb25UeXBlOiAiYWJzb2x1dGUiLAogICAgICAgICAgYWxpZ25TZWxmOiAiY2VudGVyIiwKICAgICAgICAgIHBvc2l0aW9uOiB7CiAgICAgICAgICAgIGxlZnQ6ICIyNSUiLAogICAgICAgICAgICB0b3A6ICIzMCUiCiAgICAgICAgICB9CiAgICAgICAgfX0KICAgICAgICBvbk1vdXNlRG93bj17KCkgPT4ge319CiAgICAgIC8%2BCiAgICAgIDxVaUVudGl0eQogICAgICAgIHVpVHJhbnNmb3JtPXt7CiAgICAgICAgICB3aWR0aDogJzEwMCUnLAogICAgICAgICAgaGVpZ2h0OiAzMCwKICAgICAgICAgIHBvc2l0aW9uOiB7CiAgICAgICAgICAgIHRvcDogIjEwJSIKICAgICAgICAgIH0sCiAgICAgICAgICBwb2ludGVyRmlsdGVyOiBQb2ludGVyRmlsdGVyTW9kZS5QRk1fTk9ORQogICAgICAgIH19CiAgICAgICAgdWlUZXh0PXt7CiAgICAgICAgICB2YWx1ZTonRE9FU05UIEJMT0NLIFBPSU5URVInLAogICAgICAgICAgZm9udFNpemU6MTgKICAgICAgICB9fQogICAgICAvPgogICAgICA8QnV0dG9uCiAgICAgICAgdWlUcmFuc2Zvcm09e3sgd2lkdGg6IDEwMCwgaGVpZ2h0OiA0MCwgbWFyZ2luOiA4IH19CiAgICAgICAgdmFsdWU9J1RISVMgSVMgQSBCVVRUT04nCiAgICAgICAgdmFyaWFudD0ncHJpbWFyeScKICAgICAgICBmb250U2l6ZT17MTR9CiAgICAgICAgb25Nb3VzZURvd249eygpID0%2BIHsKICAgICAgICAgIAogICAgICAgIH19CiAgICAgIC8%2BCiAgICAgPC9VaUVudGl0eT4KICA8L1VpRW50aXR5PgopKQo%3D)
2. After the world loads, hit the "Run" button
3. In the new UI panel that you will see in the middle of the screen, check that the following behaviour happens, by **clicking in every specific point shown in the image**. By "blocks pointer" I mean the pointer is not locked behind the UI (so if a UI element blocks the pointer, you won't be able to lock the pointer to move the camera, you won't be able to move the camera and you will still see the pointer)

Screen Shot 2023-10-19 at 21 14 48

### CODE REVIEW

**Comment by pravusjif** [comment]
Final protocol package will be used after [Protocol PR](https://github.com/decentraland/protocol/pull/180) is merged and before merging this PR.

**Review by anicalbano — APPROVED** [review]
🟢 Tested and reviewed by QA team

Uploading 19.10.2023_16.29.10_REC.mp4…

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UIBackground/UIBackgroundHandler.cs [hunk 1]
```diff
@@ -44,6 +44,7 @@ public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBUiB
             image.Slices = model.GetBorder().ToUnityBorder();
             image.UVs = model.Uvs.ToDCLUVs();
             image.ScaleMode = model.TextureMode.ToDCLImageScaleMode();
+            // image.canvas.pickingMode = model.PointerBlocker ? PickingMode.Position : PickingMode.Ignore;
```
Comment (davidejensen) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UIBackground/UIBackgroundHandler.cs:47]: better to just remove the line

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UIText/UiTextHandler.cs [hunk 2]
```diff
@@ -42,6 +42,7 @@ public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBUiT
             uiElement.style.color = model.GetColor().ToUnityColor();
             uiElement.style.fontSize = model.GetFontSize();
             uiElement.style.unityTextAlign = model.GetTextAlign().ToUnityTextAlign();
+            // uiElement.pickingMode = model.PointerBlocker ? PickingMode.Position : PickingMode.Ignore;
```
Comment (davidejensen) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UIText/UiTextHandler.cs:45]: same as before

**Review by davidejensen — APPROVED** [review]
approved with minor suggestion

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UIBackground/UIBackgroundHandler.cs (hunk 1 above)
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/UIBackground/UIBackgroundHandler.cs:47]: Oh yes I intended to do that but forgot, thanks!

**Pull request merged**